packages/babel-parser/src/parser/statement.ts

```diff
@@ -1208,6 +1216,8 @@ export default abstract class StatementParser extends ExpressionParser {
     if (!oldStrict) {
       this.setStrict(false);
     }
+
+    this.next();
```

Comment:
This is essentially reverting changes in https://github.com/babel/babel/pull/14980/commits/101cad5322f47907244575df1713e09d355ec31b.